### PR TITLE
[PERFORMANCE] Moment to date-fns

### PR DESCRIPTION
**Pull request by hv7214**
**Details**
Replaced `moment` with `date-fns` because of following reasons:
1. date-fns is immutable. That is it returns new Date instance from function which was run rather changing its own state.
2. date-fns is divided by function level. So only those things will be imported which are needed.

**Related issues**
https://github.com/moira-alert/moira/issues/276

**Benchmarks**
Using moment
Screenshot from 2020-02-18 02-35-59
Screenshot from 2020-02-18 02-36-12

Using date-fns
Screenshot from 2020-02-18 02-49-15
Screenshot from 2020-02-18 02-49-27

### CODE REVIEW

**Comment by sashasushko** [comment]
@hv7214 thak you for pull request. I'll review it at weekend. Maybe earlier, but I can't poromise.

**Comment by sashasushko** [comment]
@hv7214 I not forgot ;-)

**Comment by hv7214** [comment]
@sashasushko so what's say ? Does it look good to go ?

File: webpack.common.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const path = require("path");
-const MomentLocalesPlugin = require("moment-locales-webpack-plugin");
 const HtmlWebpackPlugin = require("html-webpack-plugin");
+const ContextReplacementPlugin = require("webpack").ContextReplacementPlugin;
+const supportedLocales = ["en", "ru", "es-us"];
```
Comment (sashasushko) [inline webpack.common.js]: I think `en` is enough

File: src/helpers/DateUtil.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// @flow
+import { getUnixTime } from "date-fns";
+
+function getCurrentDate() {
```
Comment (sashasushko) [inline src/helpers/DateUtil.js]: Maybe `getCurrentUTCDate`?

Comment (sashasushko) [reply src/helpers/DateUtil.js]: Even `getUTCDate`

File: src/helpers/DateUtil.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    const date = new Date();
+    const nowUtc = Date.UTC(
+        date.getUTCFullYear(),
+        date.getUTCMonth(),
+        date.getUTCDate(),
+        date.getUTCHours(),
+        date.getUTCMinutes(),
+        date.getUTCSeconds()
+    );
+    return new Date(nowUtc);
```
Comment (sashasushko) [inline src/helpers/DateUtil.js]: And I'd put `Date.UTC` inside `new Date`

File: src/helpers/DateUtil.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+function getCurrentUnixTime(): String {
```
Comment (sashasushko) [inline src/helpers/DateUtil.js]: I think it's more visibility if we'll didn't wrap `getUnixTime(getCurrentDate())` to function

Comment (sashasushko) [reply src/helpers/DateUtil.js]: Just call it as is when it needed

File: src/helpers/checkMaintenance.js [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 import { getMaintenanceCaption } from "../Domain/Maintenance";
 
 export default function checkMaintenance(maintenance: ?number): string {
     if (!maintenance) {
         return getMaintenanceCaption("off");
     }
-    const delta = maintenance - moment.utc().unix();
-    return delta <= 0 ? getMaintenanceCaption("off") : moment.duration(delta * 1000).humanize();
+    const delta = maintenance - getCurrentUnixTime();
+    return delta <= 0 ? getMaintenanceCaption("off") : formatDistance(0, delta * 1000);
```
Comment (sashasushko) [inline src/helpers/checkMaintenance.js]: But in cases like this I'll prefer on the contrary wrap code to functions

Comment (sashasushko) [reply src/helpers/checkMaintenance.js]: Because it's difficult to understand, what means `* 1000`

Comment (sashasushko) [reply src/helpers/checkMaintenance.js]: Let's write some function like `parseUnixTimestampToJS`

File: src/helpers/DateUtil.js (hunk 2 above)
Comment (hv7214) [reply src/helpers/DateUtil.js]: yeah that's sound more good

File: src/helpers/checkMaintenance.js (hunk 5 above)
Comment (hv7214) [reply src/helpers/checkMaintenance.js]: sure i'll do that : )

File: webpack.common.js (hunk 1 above)
Comment (hv7214) [reply webpack.common.js]: okay

File: src/Components/EventList/EventList.jsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 // @flow
 import * as React from "react";
-import moment from "moment";
+import { format, fromUnixTime } from "date-fns";
```
[the remark this answers is not in the document]
Comment (hv7214) [reply src/Components/EventList/EventList.jsx]: i think inline importing seems more modular, but if you say so then surely will that .

**Comment by hv7214** [comment]
@sashasushko is it good to go ? or any further changes are required ?

**Comment by sashasushko** [comment]
@hv7214 it's all okay. Thanks for good job. I need some time to add snapshot and unit tests, that auto checks pr like yours

**Comment by sashasushko** [comment]
@sorovlad посмотри, пожалуйста, код. А то я тоже тут дописывал, могу быть необъективен.

File: src/Components/Mobile/MobileMetricsListItem/MobileMetricsListItem.jsx [hunk 7]
```diff
@@ -11,6 +11,7 @@ import MobileStatusIndicator from "../MobileStatusIndicator/MobileStatusIndicato
 import roundValue from "../../../helpers/roundValue";
 import { Maintenances, type Maintenance, getMaintenanceCaption } from "../../../Domain/Maintenance";
 import cn from "./MobileMetricsListItem.less";
+import { getUTCDate } from "../../../helpers/DateUtil";
```
Comment (sorovlad) [inline src/Components/Mobile/MobileMetricsListItem/MobileMetricsListItem.jsx:14]: Here too

Comment (sashasushko) [reply src/Components/Mobile/MobileMetricsListItem/MobileMetricsListItem.jsx:14]: I'll fix it separately. Thanks!

**Pull request merged**